arbos/block_processor.go

```diff
@@ -118,7 +118,8 @@ type SequencingHooks struct {
 	TxErrors                []error
 	DiscardInvalidTxsEarly  bool
 	PreTxFilter             func(*params.ChainConfig, *types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, *arbitrum_types.ConditionalOptions, common.Address, *L1Info) error
-	PostTxFilter            func(*types.Header, *arbosState.ArbosState, *types.Transaction, common.Address, uint64, *core.ExecutionResult) error
+	PostTxFilter            func(*types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, common.Address, uint64, *core.ExecutionResult) error
+	BlockFilter             func(*types.Header, *state.StateDB, types.Transactions, types.Receipts) error
```

Comment:
Since this one has a nil check but the others don't, it might be worth commenting which must be specified and which can be nil